tests/jobserver/test_fields.py

```diff
@@ -136,3 +140,19 @@ def test_roles_field_to_python_with_empty_list():
 def test_roles_field_to_python_with_falsey_value():
     assert RolesField().to_python(None) is None
     assert RolesField().to_python([]) == []
+
+
+@pytest.mark.django_db
+def test_roles_field_with_invalid_role():
+    org = OrgFactory()
+    user = UserFactory()
+
+    class ProjectRole:
+        models = ["jobserver.models.ProjectMembership"]
+        permissions = []
+
+    msg = "Some roles could not be assigned to OrgMembership\n"
+    msg += " - ProjectRole can only be assigned to these models:\n"
+    msg += "   - jobserver.models.ProjectMembership"
+    with pytest.raises(ValueError, match=msg):
+        OrgMembershipFactory(org=org, user=user, roles=[ProjectRole])
```

Comment:
The code that's actually being tested here is obscured by the machinery of `OrgMembershipFactory`, and so it's not obvious to the casual reader what's being tested.

Reply:
Good call, added a comment to explain the test.